### PR TITLE
Ni 7699: add argument to mute info level logs

### PR DESCRIPTION
**Pull request by himsgupta**

### CODE REVIEW

File: cmd/share.go [hunk 1]
```diff
@@ -268,9 +269,13 @@ func buildShareOptions(cmd *cobra.Command) options.Options {
 	}
 
 	debug := viper.GetBool("debug")
+	disableinfologs := viper.GetBool("disableinfologs")
 	if debug != true {
 		log.SetLevel(log.InfoLevel)
 	}
+	if disableinfologs == true {
+        log.SetOutput(ioutil.Discard)
```
Comment (vikas-kph) [inline cmd/share.go]: is this formatted correctly?

**Comment by iostreamdoth** [comment]
Ideally you should change all the info levels to debug and set log level as INFO, DEBUG through command line.
Something like --log-cli-level=INFO in pytest

File: cmd/share.go (hunk 1 above)
Comment (iostreamdoth) [reply cmd/share.go]: not a good idea to do this. Use log levels to control what is printed and how it is printed. What log is not required as info change it to debug level.

**Comment by himsgupta** [comment]
Before muting logs
Untitled1

After muting:
Untitled

**Comment by himsgupta** [comment]
@iostreamdoth : 
1) here ticket is for muting all the INFO logs (since this is impacting performance on the partner side)
we are adding an optional --disableinfologs flag through command line args.

2) from the parent s3s2 repo debug flag is used to enforce log levels, 
if we dont pass debug flag , logger is set to INFO else logger is set to DEBUG (for testing) [here]
(https://github.com/tempuslabs/s3s2/blob/3adb3e8d50485ec7a779bc0532a8a1276f1bb0c5/cmd/share.go#L271)
cc: @dpalaniraja

**Comment by vikas-kph** [comment]
What happens if `debug` is set to `true` and `disableinfologs` is set to `false`. Shouldn't there be just one variable controlling the log levels

**Comment by iostreamdoth** [comment]
> @iostreamdoth :
> 
> 1. here ticket is for muting all the INFO logs (since this is impacting performance on the partner side)
>    we are adding an optional --disableinfologs flag through command line args.
> 2. from the parent s3s2 repo debug flag is used to enforce log levels,
>    if we dont pass debug flag , logger is set to INFO else logger is set to DEBUG (for testing) [here]
>    (https://github.com/tempuslabs/s3s2/blob/3adb3e8d50485ec7a779bc0532a8a1276f1bb0c5/cmd/share.go#L271
>    )
>    cc: @dpalaniraja

Including different variables to control log printing is sort of an anti-pattern. + the logic you have written just writes log to discard this will not improve performance as logs will still be written. However not printed but discarded. If you truly want to mute log level you should control log levels and not where these logs are being written. A simple change in different levels of log should do the trick.

**Comment by iostreamdoth** [comment]
Could you also provide performance improvement metrics you observed using this change?

**Comment by himsgupta** [comment]
Overall I see performace 

> Could you also provide performance improvement metrics you observed using this change?

Overall Performance metrics: ~ 8% 
Data           Timewithout DisableLog         Time with disableinfologs PercentChange
20 MB          6.48                           5.947                     = 9 %
57 MB          8.156                          7.546                     = 8.08%  

this muting logs using ioutil.Discard is similar to pointing logs /dev/null in linux, (basically writing to nowhere), 

yeah , I can add as @iostreamdoth suggested  --log-cli-level based on log level and prompting logs which are above log.PanicLevel (ignore info logs)  cc: @dpalaniraja

**Comment by iostreamdoth** [comment]
> Overall Performance metrics: ~ 8% Data Timewithout DisableLog Time with disableinfologs PercentChange 20 MB 6.48 5.947 = 9 % 57 MB 8.156 7.546 = 8.08%

I see data size is pretty small and that too both show ~.5 seconds improvement ? 

I would recommend changing the info level logs to debug level rather than imposing a fork in place where logs are written. The log levels exists for this purpose only. Let's not reinvent logging. If someone wants to view debug level logs they can set the flag true or something like that. You dont have to add a new command option. you can use debug command line option itself. Pretty much just chaanging info or infof to debug or debugf

**Comment by iostreamdoth** [comment]
you can also add [go tqdm](https://pkg.go.dev/github.com/sbwhitecap/tqdm)
this is like a progress bar if you want to have some sort of progress notification for user (not mandatory).

**Comment by dpalaniraja** [comment]
@himsgupta can we update the existing info logs  to debug logs so that we can use the existing debug parameter itself to enable/disable system io logs ? this is in reference to above [comment](https://github.com/tempuslabs/s3s2/pull/26#issuecomment-1080941815) from @vikas-kph where having debug level to true and info level to flase might produce incomplete logs.
Hence, if we could change all occurrences of `log.Infof` to `log.debugf`, we can control entire log level outputs with one existing `debug` parameter. Thoughts please ?

**Comment by himsgupta** [comment]
Untitled1

**Comment by himsgupta** [comment]
Moving the info to debug : 
Untitled

**Comment by himsgupta** [comment]
using debug flag we are g
Untitled

**Review by dpalaniraja — DISMISSED** [review]

File: aws_helpers/s3helper.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ func UploadFile(sess *session.Session, org string, aws_key string, local_path st
                 if err != nil {
                     utils.PanicIfError("Failed to upload file: ", err)
                 } else {
-                    log.Infof("File '%s' uploaded to: '%s'", file.Name(), result.Location)
+                    log.Debug("File '%s' uploaded to: '%s'", file.Name(), result.Location)
```
Comment (iostreamdoth) [inline aws_helpers/s3helper.go]: log.Debugf or log.Debug?

**Review by iostreamdoth — CHANGES_REQUESTED** [review]
you are printing %s in logs

**Comment by iostreamdoth** [comment]
could you produce screenshots with ....
/usr/bin/time go run main.go .........

**Comment by himsgupta** [comment]
Untitled

**Review by iostreamdoth — APPROVED** [review]

**Pull request merged**